### DISCUSSION TITLE
v1.17.0

### DISCUSSION THREAD

**Discussion by kyleconroy** [sqlc-dev/sqlc#2070] [Announcements]
### ## What's Changed
* codegen: Remove built-in Kotlin support by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1935
* codegen: Remove built-in Python support by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1936
* Update language-support.rst by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1937
* build(deps): bump github.com/mattn/go-sqlite3 from 1.14.15 to 1.14.16 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1913
* build(deps): bump github.com/spf13/cobra from 1.6.0 to 1.6.1 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1909
* build: Fix devcontainer by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1942
* docker: Change ENTRYPOINT to CMD by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1943
* build: Run sqlc-pg-gen via GitHub Actions by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1944
* fix: initialize generated code outside function by @akutschera in https://github.com/kyleconroy/sqlc/pull/1850
* build: Move large arrays out of functions by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1947
* mysql: Add datatype tests by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1948
* fix(engine/mysql): take into account column's charset to distinguish text/blob, (var)char/(var)binary by @mrngm in https://github.com/kyleconroy/sqlc/pull/1895
* mysql: Fix blob tests by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1949
* Add support for emitting pointer types for nullable columns by @nickbruun in https://github.com/kyleconroy/sqlc/pull/1571
* build: Fix conflicts from pointer configuration by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1950
* [PART 2] pgx v5 support (#1823) by @mcdoker18 in https://github.com/kyleconroy/sqlc/pull/1874
* feat: Add the diff command by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/1963
* [PART 3] pgx v5 support (#1823) by @mcdoker18 in https://github.com/kyleconroy/sqlc/pull/1875
* *: clean up some tiny things by @jreut in https://github.com/kyleconroy/sqlc/pull/1966
* sqlite: supported between expr (#1958) by @mcdoker18 in https://github.com/kyleconroy/sqlc/pull/1967
* Suggestion: use bufbuild to generate proto by @nickjackson in https://github.com/kyleconroy/sqlc/pull/1974
* build(deps): bump github.com/go-sql-driver/mysql from 1.6.0 to 1.7.0 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1988
* tools: regenerate scripts skips dirs that contains diff exec command by @mcdoker18 in https://github.com/kyleconroy/sqlc/pull/1987
* build(deps): bump github.com/jackc/pgtype from 1.12.0 to 1.13.0 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1978
* build(deps): bump golang from 1.19.3 to 1.19.4 by @dependabot in https://github.com/kyleconroy/sqlc/pull/1992
* build(deps): bump certifi from 2020.12.5 to 2022.12.7 in /docs by @dependabot in https://github.com/kyleconroy/sqlc/pull/1993
* plugins: Upgrade to go-wasmtime 3.0.2 by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/2009
* add version to bug template by @davideimola in https://github.com/kyleconroy/sqlc/pull/2000
* fix: the enum Value method returns correct type by @mcdoker18 in https://github.com/kyleconroy/sqlc/pull/1996
* build(deps): bump golang from 1.19.4 to 1.19.5 by @dependabot in https://github.com/kyleconroy/sqlc/pull/2016
* update uuid docs for mysql by @mikemackintosh in https://github.com/kyleconroy/sqlc/pull/2019
* Manual fix for incorrect handling of "calories" by @mpyw in https://github.com/kyleconroy/sqlc/pull/2018
* docs: Update samples for HOW-TO GUIDES by @d-tsuji in https://github.com/kyleconroy/sqlc/pull/1953
* cmd: Generate packages in parallel by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/2026
* internal/codegen: cache pattern matching compilations by @inconshreveable in https://github.com/kyleconroy/sqlc/pull/2028
* fix: Documentation for returning columns from inserted rows by @tergel-sama in https://github.com/kyleconroy/sqlc/pull/2034
* Changed documentation for json datatype. by @f1forhelp in https://github.com/kyleconroy/sqlc/pull/1997
* build(deps): bump golang from 1.19.5 to 1.20.0 by @dependabot in https://github.com/kyleconroy/sqlc/pull/2045
* fix: Add import statements even if only pointer types exist by @zaneli in https://github.com/kyleconroy/sqlc/pull/2046
* build(deps): bump github.com/jackc/pgtype from 1.13.0 to 1.14.0 by @dependabot in https://github.com/kyleconroy/sqlc/pull/2062
* fix: Search from Rexpr if not found from Lexpr by @zaneli in https://github.com/kyleconroy/sqlc/pull/2056
* build(deps): bump github.com/jackc/pgx/v4 from 4.17.2 to 4.18.0 by @dependabot in https://github.com/kyleconroy/sqlc/pull/2063
* wasm: Upgrade to wasmtime 5.0.0 by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/2065
* Only enable postgresql support when cgo is enabled as it is required by @anuraaga in https://github.com/kyleconroy/sqlc/pull/2068
* cmd/sqlc: Bump version to 1.17.0 by @kyleconroy in https://github.com/kyleconroy/sqlc/pull/2066

## New Contributors
* @mrngm made their first contribution in https://github.com/kyleconroy/sqlc/pull/1895
* @nickbruun made their first contribution in https://github.com/kyleconroy/sqlc/pull/1571
* @jreut made their first contribution in https://github.com/kyleconroy/sqlc/pull/1966
* @nickjackson made their first contribution in https://github.com/kyleconroy/sqlc/pull/1974
* @davideimola made their first contribution in https://github.com/kyleconroy/sqlc/pull/2000
* @mikemackintosh made their first contribution in https://github.com/kyleconroy/sqlc/pull/2019
* @d-tsuji made their first contribution in https://github.com/kyleconroy/sqlc/pull/1953
* @tergel-sama made their first contribution in https://github.com/kyleconroy/sqlc/pull/2034
* @f1forhelp made their first contribution in https://github.com/kyleconroy/sqlc/pull/1997
* @zaneli made their first contribution in https://github.com/kyleconroy/sqlc/pull/2046
* @anuraaga made their first contribution in https://github.com/kyleconroy/sqlc/pull/2068

**Full Changelog**: https://github.com/kyleconroy/sqlc/compare/v1.16.0...v1.17.0

<hr /><em>This discussion was created from the release <a href='https://github.com/kyleconroy/sqlc/releases/tag/v1.17.0'>v1.17.0</a>.</em>

**Comment by quangvoodoo** [comment]
@kyleconroy: generating codes with Docker doesn't work anymore with this new version
```
$ docker run --rm -v "$(pwd):/src" -w /src kjconroy/sqlc:1.17.0 generate
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "generate": executable file not found in $PATH: unknown.
```
The same command worked in v1.16.0

**Reply by kyleconroy** [reply]
@quangvoodoo I think this breaking change is expected due to https://github.com/kyleconroy/sqlc/pull/1943. Can you try running this instead?

```
$ docker run --rm -v "$(pwd):/src" -w /src kjconroy/sqlc:1.17.0 sqlc generate
```

**Reply by techschool** [reply]
@kyleconroy I tried your command but it doesn't work either:
```
❯ docker run --rm -v "$(pwd):/src" -w /src kjconroy/sqlc:1.17.0 sqlc generate
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "sqlc": executable file not found in $PATH: unknown.
```

But this one works:
```
❯ docker run --rm -v "$(pwd):/src" -w /src kjconroy/sqlc:1.17.0 /workspace/sqlc generate
```

Anyway, the docs on [this page](https://docs.sqlc.dev/en/latest/overview/install.html#docker) should be updated.

**Reply by quangvoodoo** [reply]
@kyleconroy can you update the documentation? Many people are having trouble with this because they follow the instruction in the docs and it doesn't work.

**Reply by kyleconroy** [reply]
I've reverted the change to the Dockerfile and am in the process of releasing [v1.17.1](https://github.com/kyleconroy/sqlc/releases/tag/v1.17.1)

**Reply by kyleconroy** [reply]
Whoops, looks like v1.17.1 had a build issue on Windows. Release [1.17.2](https://github.com/kyleconroy/sqlc/releases/tag/v1.17.2) with the fix.
